main/test/ca/uwaterloo/flix/language/phase/TestRedundancy.scala

```diff
@@ -808,4 +808,69 @@ class TestRedundancy extends FunSuite with TestUtils {
     compile(input, DefaultOptions).get
   }
 
+  test("UnconditionalRecursion.01") {
+    val input =
+      s"""
+         |def f(): Int =
+         |    f()
+         |""".stripMargin
+    val result = compile(input, DefaultOptions)
+    expectError[RedundancyError.UnconditionalRecursion](result)
+  }
+
+  test("UnconditionalRecursion.02") {
+    val input =
+      s"""
+         |def foo(x: Int, y: Int): _ =
+         |    foo(x + y)
+         |""".stripMargin
+    compile(input, DefaultOptions).get
```

Comment:
The compiler should have rejected this program, but there is a bug in checker return types. (You are not allowed to put _ as a return type, even if it does not complain here)

Moreover, should we even get a warning here? I am unsure.

Reply:
I've removed this case. This was meant to be a "happy" case which the compiler should allow, but I was a little too lazy to figure out what the return type should be, so I ran with this.